From b2fd697438db60889f462a4ed11f08438380b1a6 Mon Sep 17 00:00:00 2001
Message-ID: <b2fd697438db60889f462a4ed11f08438380b1a6.1787144643.git.jdenemar@redhat.com>
From: Peter Krempa <pkrempa@redhat.com>
Date: Mon, 20 Jul 2026 13:31:39 +0200
Subject: [PATCH] qemu: monitor: Remove support for 'watchdog-set-action'
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We now use 'set-action' instead.

Signed-off-by: Peter Krempa <pkrempa@redhat.com>
Reviewed-by: Ján Tomko <jtomko@redhat.com>
(cherry picked from commit 8365c6679fa65b66b85da2773547515a40ae1a8f)

https://redhat.atlassian.net/browse/RHEL-242546
---
 src/qemu/qemu_monitor.c      | 12 ------------
 src/qemu/qemu_monitor.h      |  4 ----
 src/qemu/qemu_monitor_json.c | 22 ----------------------
 src/qemu/qemu_monitor_json.h |  5 -----
 4 files changed, 43 deletions(-)

diff --git a/src/qemu/qemu_monitor.c b/src/qemu/qemu_monitor.c
index 3d7477c01c..5ecee978f9 100644
--- a/src/qemu/qemu_monitor.c
+++ b/src/qemu/qemu_monitor.c
@@ -4006,18 +4006,6 @@ qemuMonitorMemoryDeviceSizeChangeFree(qemuMonitorMemoryDeviceSizeChangePtr info)
 }
 
 
-int
-qemuMonitorSetWatchdogAction(qemuMonitor *mon,
-                             const char *action)
-{
-    VIR_DEBUG("watchdogAction=%s", action);
-
-    QEMU_CHECK_MONITOR(mon);
-
-    return qemuMonitorJSONSetWatchdogAction(mon, action);
-}
-
-
 /**
  * qemuMonitorBlockdevCreate:
  * @mon: monitor object
diff --git a/src/qemu/qemu_monitor.h b/src/qemu/qemu_monitor.h
index bbe400003a..671930fe5f 100644
--- a/src/qemu/qemu_monitor.h
+++ b/src/qemu/qemu_monitor.h
@@ -1642,10 +1642,6 @@ qemuMonitorSetBlockThreshold(qemuMonitor *mon,
                              const char *nodename,
                              unsigned long long threshold);
 
-int
-qemuMonitorSetWatchdogAction(qemuMonitor *mon,
-                             const char *action);
-
 int
 qemuMonitorBlockdevCreate(qemuMonitor *mon,
                           const char *jobname,
diff --git a/src/qemu/qemu_monitor_json.c b/src/qemu/qemu_monitor_json.c
index 45f690d9e2..a98cf70a41 100644
--- a/src/qemu/qemu_monitor_json.c
+++ b/src/qemu/qemu_monitor_json.c
@@ -7826,28 +7826,6 @@ qemuMonitorJSONSetBlockThreshold(qemuMonitor *mon,
 }
 
 
-int
-qemuMonitorJSONSetWatchdogAction(qemuMonitor *mon,
-                                 const char *action)
-{
-    g_autoptr(virJSONValue) cmd = NULL;
-    g_autoptr(virJSONValue) reply = NULL;
-
-    if (!(cmd = qemuMonitorJSONMakeCommand("watchdog-set-action",
-                                           "s:action", action,
-                                           NULL)))
-        return -1;
-
-    if (qemuMonitorJSONCommand(mon, cmd, &reply) < 0)
-        return -1;
-
-    if (qemuMonitorJSONCheckError(cmd, reply) < 0)
-        return -1;
-
-    return 0;
-}
-
-
 int
 qemuMonitorJSONBlockdevCreate(qemuMonitor *mon,
                               const char *jobname,
diff --git a/src/qemu/qemu_monitor_json.h b/src/qemu/qemu_monitor_json.h
index b418f70048..b07e54c2c5 100644
--- a/src/qemu/qemu_monitor_json.h
+++ b/src/qemu/qemu_monitor_json.h
@@ -629,11 +629,6 @@ qemuMonitorJSONSetBlockThreshold(qemuMonitor *mon,
                                  unsigned long long threshold)
     ATTRIBUTE_NONNULL(1) ATTRIBUTE_NONNULL(2);
 
-int
-qemuMonitorJSONSetWatchdogAction(qemuMonitor *mon,
-                                 const char *action)
-    ATTRIBUTE_NONNULL(1);
-
 int
 qemuMonitorJSONBlockdevCreate(qemuMonitor *mon,
                               const char *jobname,
-- 
2.55.0
